api/insight/apiroutes.go

```diff
@@ -287,27 +310,136 @@ func (c *insightApiContext) getTransactions(w http.ResponseWriter, r *http.Reque
 }
 
 func (c *insightApiContext) getAddressesTxn(w http.ResponseWriter, r *http.Request) {
-	address := m.GetAddressCtx(r)
-	count := m.GetCountCtx(r)
-	offset := m.GetOffsetCtx(r)
+	address := c.GetAddressCtx(r)         // Required
+	noAsm := c.GetNoAsmCtx(r)             // Optional
+	noScriptSig := c.GetNoScriptSigCtx(r) // Optional
+	noSpent := c.GetNoSpentCtx(r)         // Optional
+	from, ok := c.GetFromCtx(r)           // Optional
 
-	if address == "" {
-		http.Error(w, http.StatusText(422), 422)
-		return
+	if !ok {
+		from = 0
+	}
+
+	to, ok := c.GetToCtx(r) // Optional
+	if !ok {
+		to = from + 10
 	}
 
-	addresses := strings.Split(address, ",")
+	addresses := []string{}
+
+	// If no address present we need to stop.  Allow Addresses to be single or
+	// multiple separated by a comma.
+	if address != "" {
+		if strings.Contains(address, ",") {
+			addresses = strings.Split(address, ",")
+		} else {
+			addresses = []string{address}
+		}
+	} else {
```

Comment:
I suggest to check this right after `GetAddressCtx`, then down here you don't need any checks, just `addresses := strings.Split(address, ",")`